clang/lib/AST/Type.cpp

```diff
@@ -2611,7 +2613,7 @@ bool Type::isLiteralType(const ASTContext &Ctx) const {
   // As an extension, Clang treats vector types and complex types as
   // literal types.
   if (BaseTy->isScalarType() || BaseTy->isVectorType() ||
-      BaseTy->isAnyComplexType())
+      BaseTy->isAnyComplexType() || BaseTy->isDecimalFloatType())
```

Comment:
I think DFP types should be considered scalar types; the check for `isScalarType()` should therefore suffice here.

(note for myself: this is as far as I got with review before running out of time for the day)